### DISCUSSION TITLE
Email sending stopped working after 1.2 upgrade

### DISCUSSION THREAD

**Discussion by dpiedra-isg** [reliatec-gmbh/LibreClinica#356] [Q&A] [answered]
It seems that the email sending stopped working after upgrading LC to 1.2 version.

We are using gmail as smtp, and the datainfo.properties block is:

mailHost=smtp.gmail.com
mailPort=465
mailProtocol=smtps
mailUsername=email@gmail.com
mailPassword=password
mailSmtpAuth=true
mailSmtpStarttls.enable=false
mailSmtpsAuth=true
mailSmtpsStarttls.enable=false
mailErrorMsg=email@gmail.com
mailSmtpConnectionTimeout=1000

The server can reach successfully gmail servers, even send emails (tested with another tool).

Also have detected that when changing logLevel to "debug", LC app does not deploy correctly, and we get a TOMCAT error.

**Comment by oba-reliatec** [comment]
Does email work again when you revert to the old LC.war without changing anything else? Which LC version did you use before when it worked? What is your Java version (please be exact here, `java -version`) and operating system?

**Reply by dpiedra-isg** [reply]
In fact, we tried, but the whole LC stopped working, but because other issues! In the end, we did a fresh start of the 1.1 and everything fine again. We will set a test environment, so we can test it deeply. Thanks for the answer!

**Comment by toskrip** [comment, marked as answer]
Just for completeness so that this does not stay unanswered. After investigation this turned out not to be only LC 1.2 related problem but general problem with security JDK updates. Depending on default configuration of how JDK in packaged and distributed to specific linux distributions more people can hit this (especially in Debian). Older TLS protocols seem to be disabled now see e.g. this issue:

https://stackoverflow.com/questions/67899129/postfix-and-openjdk-11-no-appropriate-protocol-protocol-is-disabled-or-cipher

Upgrading the mailing library seem to activate the default usage of newer TLS protocol so  the LC community published the advisory how to upgrade library without need of waiting for release here:

https://libreclinica.org/archive/mail_update_warning_20221103.html

However we also made a ticket and in current developer branch the newer library is used by default:

https://github.com/reliatec-gmbh/LibreClinica/issues/357

best

Tomas

**Reply by dpiedra-isg** [reply]
This makes sense. In the end we did a fresh install of the 1.1 version, but keeping your observations in mind, we will test it! Thanks for your feedback!
